### PR TITLE
log difference in message edits

### PR DESCRIPTION
**Pull request by JulianVennen**

### CODE REVIEW

File: features/messageUpdate/edited.js [hunk 1]
```diff
@@ -9,13 +10,27 @@ exports.event = async (database, old, newMsg) => {
...
+    if (part.added) {
+      formatted += `__${part.value}__`;
+    }
+    else if (part.removed) {
+      formatted += `~~${part.value}~~ `;
+    }
+    else {
+      formatted += part.value;
+    }
+  }
```
Comment (KurtThiemann) [inline features/messageUpdate/edited.js:26]:
```suggestion
  let formatted = '';
  let maxLength = 1991;
  for (let part of diff) {
    let maxPartLength = maxLength - formatted.length;
    if (part.added) {
      formatted += `__${part.value.substr(0, maxPartLength - 4)}__`;
    }
    else if (part.removed) {
      formatted += `~~${part.value.substr(0, maxPartLength - 4)}~~ `;
    }
    else {
      formatted += part.value.substr(0, maxPartLength);
    }
    if(maxLength - formatted.length <= 0){
      break;
    }
  }
```

File: features/messageUpdate/edited.js [hunk 2]
```diff
@@ -9,13 +10,27 @@ exports.event = async (database, old, newMsg) => {
...
+
   let embed = new Discord.MessageEmbed()
     .setColor(util.color.orange)
     .setAuthor(`Message by ${old.author.username}#${old.author.discriminator} in #${old.channel.name} was edited`,old.author.avatarURL())
-    .addFields([
-      {name: 'Before:', value: old.content.substring(0,1024)},
-      {name: 'After:', value: newMsg.content.substring(0,1024)}
-    ])
+    .setDescription(
+      'Changes: ' + formatted.substring(0,1991)
```
Comment (KurtThiemann) [inline features/messageUpdate/edited.js:32]:
```suggestion
      'Changes: ' + formatted
```

**Review by KurtThiemann — CHANGES_REQUESTED** [review]
Check the length for each individual diff part to prevent cutting of formatting codes

File: features/messageUpdate/edited.js [hunk 3]
```diff
@@ -9,13 +10,37 @@ exports.event = async (database, old, newMsg) => {
...
 
+  let diff = jsdiff.diffWords(old.content, newMsg.content);
+
+  let formatted = '';
+  let maxLength = 1985;
+  for (let part of diff) {
+
+    //escape formatting in message
+    part.value = part.value.replace(/[\_]/g,'\\_');
+    part.value = part.value.replace(/[\~]/g,'\\~');
```
Comment (matthi4s) [inline features/messageUpdate/edited.js:21]:
```suggestion
    part.value = part.value.replace(/([_~])/g,'\\$1');
```

**Review by KurtThiemann — APPROVED** [review]

File: features/messageUpdate/edited.js [hunk 4]
```diff
@@ -9,13 +10,37 @@ exports.event = async (database, old, newMsg) => {
...
   }
 
+  let diff = jsdiff.diffWords(old.content, newMsg.content);
+
+  let formatted = '';
+  let maxLength = 1985;
+  for (let part of diff) {
+
+    //escape formatting in message
+    part.value = part.value.replace(/[\_]/g,'\\_');
```
Comment (matthi4s) [inline features/messageUpdate/edited.js:20]:
```suggestion

```

**Review by matthi4s — CHANGES_REQUESTED** [review]
Escape existing formatting

File: features/messageUpdate/edited.js [hunk 5]
```diff
@@ -9,13 +10,36 @@ exports.event = async (database, old, newMsg) => {
...
+
   let embed = new Discord.MessageEmbed()
     .setColor(util.color.orange)
     .setAuthor(`Message by ${old.author.username}#${old.author.discriminator} in #${old.channel.name} was edited`,old.author.avatarURL())
-    .addFields([
-      {name: 'Before:', value: old.content.substring(0,1024)},
-      {name: 'After:', value: newMsg.content.substring(0,1024)}
-    ])
+    .setDescription(
+      'Changes: ' + formatted
```
Comment (matthi4s) [inline features/messageUpdate/edited.js:41]:
```suggestion
      formatted
```

**Review by matthi4s — CHANGES_REQUESTED** [review]

**Review by matthi4s — APPROVED** [review]

**Pull request merged**